src/types/function.rs

```diff
@@ -102,11 +102,16 @@ impl PyCFunction {
     ///         let i = args.extract::<(i64,)>()?.0;
     ///         Ok(i+1)
     ///     };
-    ///     let add_one = types::PyCFunction::new_closure(add_one, py).unwrap();
+    ///     let add_one = types::PyCFunction::new_closure(add_one, "add_one", "add one to integer", py).unwrap();
     ///     py_run!(py, add_one, "assert add_one(42) == 43");
     /// });
     /// ```
-    pub fn new_closure<F, R>(f: F, py: Python<'_>) -> PyResult<&PyCFunction>
+    pub fn new_closure<'a, F, R>(
```

Comment:
While changing the signature, should be move `py` to the beginning? I think that's where it is in most (?) other APIs.

Reply:
I think it is also customary to move closure arguments to the end as this improves readability of the call sites.